README.md

```diff
@@ -0,0 +1,496 @@
+
+```coffeescript
+[1..3]
+  .map((x) -> x * x)
+  .concat([10..12])
+  .filter((x) -> x < 11)
+  .reduce((x, y) -> x + y)
+```
+
+When calling functions, choose to omit or include parentheses in such a way that optimizes for readability. Keeping in mind that "readability" can be subjective, the following examples demonstrate cases where parentheses have been omitted or included in a manner that the community deems to be optimal:
+
+```coffeescript
+baz 12
+
+brush.ellipse x: 10, y: 20 # Braces can also be omitted or included for readability
+
+foo(4).bar(8)
+
+obj.value(10, 20) / obj.value(20, 10)
+
+print inspect value
+
+new Tag(new Value(a, b), new Arg(c))
+```
+
+You will sometimes see parentheses used to group functions (instead of being used to group function parameters). Examples of using this style (hereafter referred to as the "function grouping style"):
+
+```coffeescript
+($ '#selektor').addClass 'klass'
+
+(foo 4).bar 8
+```
+
+This is in contrast to:
+
+```coffeescript
+$('#selektor').addClass 'klass'
+
+foo(4).bar 8
+```
+
+In cases where method calls are being chained, some adopters of this style prefer to use function grouping for the initial call only:
+
+```coffeescript
+($ '#selektor').addClass('klass').hide() # Initial call only
+(($ '#selektor').addClass 'klass').hide() # All calls
+```
+
+The function grouping style is not recommended. However, **if the function grouping style is adopted for a particular project, be consistent with its usage.**
+
+<a name="strings"></a>
+## Strings
+
+Use string interpolation instead of string concatenation:
+
+```coffeescript
+"this is an #{adjective} string" # Yes
+"this is an " + adjective + " string" # No
+```
+
+Prefer single quoted strings (`''`) instead of double quoted (`""`) strings, unless features like string interpolation are being used for the given string.
+
+<a name="conditionals"></a>
+## Conditionals
+
+Favor `unless` over `if` for negative conditions.
+
+Instead of using `unless...else`, use `if...else`:
+
+```coffeescript
+  # Yes
+  if true
+    ...
+  else
+    ...
+
+  # No
+  unless false
+    ...
+  else
+    ...
+```
+
+Multi-line if/else clauses should use indentation:
+
+```coffeescript
+  # Yes
+  if true
+    ...
+  else
+    ...
+
+  # No
+  if true then ...
+  else ...
+```
+
+<a name="looping_and_comprehensions"></a>
+## Looping and Comprehensions
+
+Take advantage of comprehensions whenever possible:
+
+```coffeescript
+  # Yes
+  result = (item.name for item in array)
+
+  # No
+  results = []
+  for item in array
+    results.push item.name
+```
+
+To filter:
+
+```coffeescript
+result = (item for item in array when item.name is "test")
+```
+
+To iterate over the keys and values of objects:
+
+```coffeescript
+object = one: 1, two: 2
+alert("#{key} = #{value}") for key, value of object
```

Comment:
I'm not a fan of this loop-statement-as-suffix style, and it's inconsistent with JS.